File: src/main/java/ru/isntrui/lb/models/Invite.java
package ru.isntrui.lb.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Table(name = "invites")
@Data
@EqualsAndHashCode(of = {"id"})
public class Invite {

    @Id
    @GeneratedValue
    private Long id;

    @Column(unique = true, nullable = false)
    private String email;

    private String code;

    @ManyToOne
    private User madeBy;

    private Boolean madeOn;
}

File: src/main/java/ru/isntrui/lb/models/Invite.kt
package ru.isntrui.lb.models

import jakarta.persistence.*
import jakarta.validation.constraints.Email
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime

@Entity
@Table(name = "invites")
open class Invite(
    @Column(unique = true, nullable = false)
    @Email
    open var email: String? = null,

    @Column(nullable = false)
    open var code: String = ""
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(name = "made_on", nullable = false)
    @CreationTimestamp
    open lateinit var madeOn: LocalDateTime

    @Column(nullable = false)
    open var isUsed: Boolean = false

    constructor() : this(null, "")
}